### PR TITLE
Feat Mentions

### PR DESCRIPTION
**Pull request by Amour1688**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/mentions/index.jsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      this.setState({
+        focused: true,
+      });
+    },
+    onChange(val) {
+      this.$emit('change', val);
+    },
+    getNotFoundContent(renderEmpty) {
+      const h = this.$createElement;
+      const { notFoundContent } = this.$props;
```
Comment (zkwolf) [inline components/mentions/index.jsx]: 这里用getComponentFromProp(this, 'notFoundContent')吧，兼容slot

Comment (Amour1688) [reply components/mentions/index.jsx]: done

Comment (Amour1688) [reply components/mentions/index.jsx]: > 这里用getComponentFromProp(this, 'notFoundContent')吧，兼容slot

done

File: components/mentions/index.jsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    event: 'change',
+  },
+  Option: { ...Option, name: 'AMentionsOption' },
+  getMentions,
+  props: initDefaultProps(
+    {
+      ...mentionsProps,
+      loading: PropTypes.bool,
+    },
+    defaultProps,
```
Comment (zkwolf) [inline components/mentions/index.jsx]: 这里我感觉没有必要初始化prop值，notFoundContent其实不该给默认值，其他组件好像都统一默认没值用empty组件展示了

Comment (Amour1688) [reply components/mentions/index.jsx]: 好像是的，我改一下

File: components/vc-mentions/src/mentionsProps.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  prefix: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
+  prefixCls: PropTypes.string,
+  value: PropTypes.string,
+  defaultValue: PropTypes.string,
+  disabled: PropTypes.bool,
+  notFoundContent: PropTypes.any,
+  split: PropTypes.string,
+  transitionName: PropTypes.string,
+  placement: PropTypes.oneOf(PlaceMent),
+  rows: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```
Comment (Amour1688) [inline components/vc-mentions/src/mentionsProps.js]: rows 本身是textarea的属性，感觉不需要加在`mentionsProps` 里面，但是 `initDefaultProps`  这个方法，设置rows时，发现没有定义在 `mentionsProps ` 会抛异常, @tangjinzhou 大佬评估下

```js
const initDefaultProps = (propTypes, defaultProps) => {
  Object.keys(defaultProps).forEach(k => {
    if (propTypes[k]) {
      propTypes[k].def && (propTypes[k] = propTypes[k].def(defaultProps[k]));
    } else {
      throw new Error(`not have ${k} prop`);
    }
  });
  return propTypes;
};
```

**Comment by zkwolf** [comment]
mentions下拉框出现以后没办法用键盘上下调整

**Comment by Amour1688** [comment]
> mentions下拉框出现以后没办法用键盘上下调整

done

File: components/vc-mentions/src/DropdownMenu.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          },
+        }}
+      >
+        {options.map((option, index) => {
+          const { value, disabled, children } = option;
+          return (
+            <MenuItem
+              key={value}
+              disabled={disabled}
+              onMouseEnter={() => {
```
Comment (tangjinzhou) [inline components/vc-mentions/src/DropdownMenu.jsx]: `onMouseEnter` to `onMouseenter`

File: components/vc-mentions/src/Mentions.jsx [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+      }
+    });
+  },
+  methods: {
+    triggerChange(value) {
+      const props = getOptionProps(this);
+      if (!('value' in props)) {
+        this.setState({ _value: value });
+      } else {
+        this.$forceUpdate();
```
Comment (tangjinzhou) [inline components/vc-mentions/src/Mentions.jsx]: 需要吗

File: components/vc-mentions/src/Mentions.jsx [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+      'split',
+      'children',
+      'validateSearch',
+      'filterOption',
+    ]);
+
+    const options = measuring ? this.getOptions() : [];
+
+    return (
+      <div class={prefixCls} style={getStyle(this)}>
```
Comment (tangjinzhou) [inline components/vc-mentions/src/Mentions.jsx]: style 不需要，会自动挂载

File: components/mentions/index.jsx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+        prefixCls,
+        notFoundContent: this.getNotFoundContent(renderEmpty),
+        ...otherProps,
+        disabled,
+        filterOption: this.getFilterOption(),
+        getPopupContainer: getPopupContainer || getContextPopupContainer,
+        children: this.getOptions(),
+      },
+      class: mergedClassName,
+      attrs: { rows: 1, ...this.$attrs },
```
[the remark this answers is not in the document]
Comment (Amour1688) [reply components/mentions/index.jsx]: rows 设置 textarea 的属性

File: components/vc-mentions/src/Mentions.jsx (hunk 5 above)
Comment (Amour1688) [reply components/vc-mentions/src/Mentions.jsx]: 为了做受控

**Pull request merged**